signimage/README.md

```diff
@@ -0,0 +1,50 @@
+A utility to pull down an image, extract named kernel modules from it, sign them with the provided keys, and add them back in as a new layer, then upload that new image under a new tag.
+
+It operates as a wrapper around the sign-file binary provided as part of the kernel-devel package (a wrapper rather than a reimplementation to ensure its bug-for-bug compatabile, rather then having awhole new set of bugs of its own). sign-file is distributed as part of the kernel source so in theory is kernel version specific but in reality it hasn't changed to 5+ years, and for kernel modules to be whitelisted across major RHEL versions the signing format also has to be compatable so this is not a major concer.
+
+Configuration is done via command line switches or failing that via environment variables
```

Comment:
Why do we need both switches and env variables? Assuming that KMM's code is the only one consuming that binary, won't it be easier to maintain a single method of passing arguments?

Reply:
> Why do we need both switches and env variables? Assuming that KMM's code is the only one consuming that binary, won't it be easier to maintain a single method of passing arguments?

Agreed.

Reply:
--help works and produces a usage message, its provided automagically by the flag package.

I'm not assuming this is only used by KMM I think it should at least be possible to use it standalone in some non-KMM build pipeline and supporting environment variables only really adds a couple of lines and actually lets you create defaults taht you can then override with args, so I think the flexability is worth it.

Reply:
If we're only going for one then switches makes more sense, I just like the envars as fallback options. However this is not red line for me, and the discussion is taking more effort than they're worth so I've removed them.  If anyone cares they can always be added back in later.